docs/adr/2022-12-05_002-version-serialization-schemes.md

```diff
@@ -0,0 +1,43 @@
+---
+slug: 2
+title: |
+  2. Versioned Serialization Schemes
+authors: [Jared Corduan]
+tags: [Accepted]
+---
+
+## Status
+
+Accepted
+
+## Context
+
+The cardando ledger used CBOR to serialize the blocks and transaction on the network.
+When we encounter a problem with our deserializers, such as in #3003, #2965 and #2444,
+it can be very difficult to implement a fix.
+It is difficult because we can only fix such issues during a hard fork,
+and leading up to the hard fork we must maintain two serializations for the same type in order
+to not cause unintended network splitting
+(the problematic version must be used before the hard fork, and the fixed version is used afterwards).
+This can be especially tricky with the FromCBOR typeclass,
+since it is not always easy to search for where all the problematic uses are located.
+
+This issue is captured in #3014.
+
+## Decision
+
+Create a new typeclass to replace ToCBOR and FromCBOR which depends not only on the type
+but also on the protocol version (or, alternatively, the ledger era).
+The first instances will all be implemented identically to our current
+ToCBOR and FromCBOR instances for all protocol versions up to the current one.
+
+## Consequences
+
+This change effects downstream components that rely on the ledger serializers,
```

Comment:
Perhaps `affects` would be better ?

Reply:
:facepalm: :laughing: you'd think I could get that one correct by now... thank you for saving me future embarrassment! :grin: